soc/riscv/riscv-ite/it8xxx2/soc.c

```diff
@@ -176,6 +176,26 @@ BUILD_ASSERT(CONFIG_FLASH_INIT_PRIORITY < CONFIG_IT8XXX2_PLL_SEQUENCE_PRIORITY,
 	"CONFIG_FLASH_INIT_PRIORITY must be less than CONFIG_IT8XXX2_PLL_SEQUENCE_PRIORITY");
 #endif /* CONFIG_SOC_IT8XXX2_PLL_FLASH_48M */
 
+#ifdef CONFIG_SOC_IT8XXX2_CPU_IDLE_ALLOWED
+/* Preventing CPU going into idle mode during command queue. */
+static atomic_t cpu_idle_disabled;
+
+void chip_permit_idle(int port)
+{
+	atomic_clear_bit(&cpu_idle_disabled, port);
+}
+
+void chip_block_idle(int port)
+{
+	atomic_set_bit(&cpu_idle_disabled, port);
+}
+
+uint32_t cpu_idle_not_allowed(void)
+{
+	return cpu_idle_disabled;
```

Comment:
Use `atomic_get` for correctness, and suggest returning a `bool`.

Reply:
Done.